eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/ContextualClientProvider.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Generator.Management.Models;
+using Azure.Generator.Management.Snippets;
+using Azure.Generator.Management.Utilities;
+using Azure.Generator.Management.Visitors;
+using Azure.ResourceManager;
+using Microsoft.TypeSpec.Generator.Expressions;
+using Microsoft.TypeSpec.Generator.Input;
+using Microsoft.TypeSpec.Generator.Input.Extensions;
+using Microsoft.TypeSpec.Generator.Primitives;
+using Microsoft.TypeSpec.Generator.Providers;
+using Microsoft.TypeSpec.Generator.Snippets;
+using Microsoft.TypeSpec.Generator.Statements;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using static Microsoft.TypeSpec.Generator.Snippets.Snippet;
+
+namespace Azure.Generator.Management.Providers
+{
+    internal abstract class ContextualClientProvider : TypeProvider
+    {
+        internal const string ValidateResourceIdMethodName = "ValidateResourceId";
+        internal const string TryGetApiVersionMethodName = "TryGetApiVersion";
+
+        private readonly RequestPathPattern _contextualRequestPattern;
+
+        protected ContextualClientProvider(RequestPathPattern contextualRequestPattern) : base()
+        {
+            _contextualRequestPattern = contextualRequestPattern;
+        }
```

Comment:
since this is only used privately, could just turn use this private field and populate it in the constructor? it is needed for this class to function anyway.